### ISSUE TITLE
Game - find by FEN

### ISSUE THREAD

**Issue by mbutsykin** [yo35/kokopu#26]
Can we have find by "FEN" method for game? similar to `findById`?

Mb even something more generic - have some method in game / variation that returns all existing nodes (including child variations of nodes) as a JS array so native functions like `map/filter/find` etc. are availiable? I'm not sure about memory usage of this mb it's better to stay with existing data structure like bidirectional list, so instead of it - implement own `filter/map/find(predicate)` methods?

**Comment by yo35** [comment]
I've just released a new version of the lib (2.7.0), with a new method `Game#nodes(withSubVariations)` that returns all the nodes of a game (including those in sub-variations in the boolean flag `withSubVariations` is true) in an array. Then it becomes possible to implement what you want as follows:
```
const posToSearch = new kokopu.Position(fenToSearch);
const result = game.nodes(true).filter(node => kokopu.Position.isEqual(node.position(), posToSearch));
```

`kokopu.Position.isEqual(..)` is also something new in 2.7.0.

**Comment by mbutsykin** [comment]
Awesome! Thank you a lot
